src/js/utils/DSLUpdateUtil.js

```diff
@@ -197,11 +204,29 @@ const DSLUpdateUtil = {
    *
    * @param {DSLExpression} expression - The expression to update
    * @param {Array} nodes - The node(s) to append
-   * @param {DSLCombinerTypes} combiner - The combiner operation to use
+   * @param {DSLCombinerTypes} [newCombiner] - The combiner for first node
+   * @param {DSLCombinerTypes} [itemCombiner] - The combiner beteen nodes
    * @returns {DSLExpression} expression - The updated expression
    */
-  applyAdd(expression, nodes, combiner=DSLCombinerTypes.AND) {
-    let expressionUpdate = nodes.reduce(({value, offset}, node) => {
+  applyAdd(expression, nodes, newCombiner=DSLCombinerTypes.AND,
```

Comment:
I think I would prefer this here to keep function declaration in one line:
```js
applyAdd(expression, nodes, newCombiner, itemCombiner) {
  newCombiner = newCombiner || DSLCombinerTypes.AND;
  itemCombiner = itemCombiner || DSLCombinerTypes.AND;
```

But it's up to you